File: CullinaryBookSite/ClientApp/src/app/Services/user.service.ts
import { Injectable } from "@angular/core";
import { Observable } from "rxjs";
import { HttpClient } from "@angular/common/http";
import { LikeDto } from "../Dto/likeDto";
import { FavoriteDto } from "../Dto/favoriteDto";

@Injectable()
export class UserService {
    private readonly apiUrl: string = '/api/user';

    constructor(private httpClient: HttpClient) {
    }

    public getLikes(userId: number) : Observable<LikeDto[]>
    {
        return this.httpClient.get<LikeDto[]>(`${this.apiUrl}/${userId}/likes`);
    }

    public getFavorites(userId: number) : Observable<FavoriteDto[]>
    {
        return this.httpClient.get<FavoriteDto[]>(`${this.apiUrl}/${userId}/favorites`);
    }
}

File: Api/Services/UserService.cs
using Domain.RecipeEntity;
using Domain.UoW;
using Api.Dto;
using Api.Services.Mappers;

namespace Api.Services
{
    public class UserService : IUserService
    {
        private readonly IUserRepository _userRepository;
        private readonly IUnitOfWork _unitOfWork;

        public UserService(IUserRepository repository, IUnitOfWork unitOfWork)
        {
            _userRepository = repository;
            _unitOfWork = unitOfWork;
        }

        public async Task<List<LikeDto>> GetLikes(int userId)
        {
            List<Like> likes = await _userRepository.GetLikes(userId);

            return likes.ConvertAll(l => l.MapToLikeDto());
        }

        public async Task<List<FavoriteDto>> GetFavorites(int userId)
        {
            List<Favorite> favorites = await _userRepository.GetFavorites(userId);

            return favorites.ConvertAll(f => f.MapToFavoriteDto());
        }
    }
}
